node/server.js

```diff
@@ -378,11 +378,20 @@ function init_game() {
 					create_instance("d_b1");
 					create_instance("d_a1");
 					create_instance("d_a2");
-					server_bfs("crypt");
-					server_bfs("winter_instance");
-					server_bfs("tomb");
-					server_bfs("dungeon0");
-					server_bfs("cgallery");
+
+					for (const name in G.maps) {
+						const gMap = G.maps[name];
+						if (gMap.ignore) {
+							continue;
+						}
+
+						const hasNpcs = gMap.npcs && gMap.npcs.length > 0;
+						const hasMonsters = gMap.monsters && gMap.monsters.length > 0;
+						if (gMap.instance && (hasNpcs || hasMonsters)) {
+							// running this is important for instances, so that npcs and monsters can navigate / move
+							server_bfs(name);
+						}
+					}
```

Comment:
I would like to see the full list of what this loop will run the bfs for, please! Can you edit the PR description with the info? Thanks!

Reply:
I'll update the description when I'm on my computer, as far as I remember this run of bfs is only needed if precompute.js has not been run. But I'll verify it.

Reply:
The code is a little different on my own branches where I've added more debug output but this part returns early if data has been precomputed in `server_bfs`
https://github.com/kaansoral/adventureland/blob/f39039c383e588e7b4b8b6fcf24793eb98fa71f6/node/server_functions.js#L4042-L4046

Reply:
I've updated the description, I have additional debug statements in my branch for my server, I could include them in the PR as well if you think so. @Telokis feel free to mark this conversation as resolved if you think it is